Git/commit.sublime-syntax

```diff
@@ -0,0 +1,81 @@
+%YAML 1.2
+---
+# Highlight regular git commits, merge commits, and tags.
+
+name: Git Commit
+file_extensions:
+  - COMMIT_EDITMSG
+  - MERGE_MSG
+  - TAG_EDITMSG
+scope: source.gitcommit
+
+contexts:
+  prototype:
+    - include: comments
+
+  main:
+    - include: commit-message
+
+  branch-line:
+    - match: \bOn branch\s+(.*)
+      captures:
+        1: support.type.branch-name.gitcommit
+
+  change:
+    - match: (new file|modified|deleted)\s*(:)\s*(.*)
+      captures:
+        1: keyword.change-list.change.gitcommit
+        2: punctuation.separator.change.gitcommit
+        3: keyword.change-list.file.gitcommit
+
+  change-list:
+    - include: change-list-header
+    - include: change
+
+  change-list-header:
+    - match: '\bChanges to be committed\s*(:)'
+      scope: keyword.change-list.heading.gitcommit
+      captures:
+        1: punctuation.separator.change-list.gitcommit
+
+  comments:
+    - match: '#'
+      scope: punctuation.definition.comment.gitcommit
+      push:
+        - meta_scope: comment.line.gitcommit
+        - match: \n
+          pop: true
+        - include: change-list
+        - include: branch-line
+        - include: date-line
+
+  commit-message:
+    # this absurd regex is necessary to:
+    #   1. match comments in the subject line
+    #   2. only style the first 50 chars in the subject line
+    - match: ([^#\n]{,50})[^#\n]*((#).*$)?
```

Comment:
As far as I can see from a couple of `git commit -m "My message # with hash # tags #"` a commit message can contain `#` anywhere without matching them as comment. Therefore I am wondering what made you create this regex?

I'd rather check for the first occurrence of `#` at the beginning of a line and expect comments only from this point on as a commit message should not be split by comments.


Furthermore why do you style the 50 characters only?

Reply:
1) I only styled the first 50 characters because this is what vim does. I was trying to replicate vim's behavior as closely as possible. I believe 50 character subject lines are a git best practice that vim tries to encourage.

2) Good catch about `#` not being a comment unless it's the first character in the line! That's a big bug.